src/test/java/org/jabref/gui/externalfiles/ImportHandlerTest.java

```diff
@@ -54,4 +102,120 @@ void handleBibTeXData() {
 
         assertEquals(List.of(expected), bibEntries.stream().toList());
     }
+
+    @Test
+    void cleanUpEntryTest() {
+        BibEntry entry = new BibEntry().withField(StandardField.AUTHOR, "Clear Author");
+        BibEntry cleanedEntry = importHandler.cleanUpEntry(bibDatabaseContext, entry);
+
+        Optional<String> authorOptional = cleanedEntry.getField(StandardField.AUTHOR);
+
+        assertEquals(Optional.of("Clear Author"), authorOptional);
+    }
+
+    @Test
+    void findDuplicateTest() {
+        // Assume there's no duplicate initially
+        assertFalse(importHandler.findDuplicate(bibDatabaseContext, testEntry).isPresent());
+    }
+
+    @Test
+    void handleDuplicatesKeepRightTest() {
+        // Arrange
+        BibEntry duplicateEntry = new BibEntry(StandardEntryType.Article)
+                .withCitationKey("Duplicate2023")
+                .withField(StandardField.AUTHOR, "Duplicate Author");
+
+        BibDatabase bibDatabase = bibDatabaseContext.getDatabase();
+        bibDatabase.insertEntry(duplicateEntry); // Simulate that the duplicate entry is already in the database
+
+        DuplicateDecisionResult decisionResult = new DuplicateDecisionResult(DuplicateResolverDialog.DuplicateResolverResult.KEEP_RIGHT, null);
+        importHandler = Mockito.spy(new ImportHandler(
+                bibDatabaseContext,
+                preferencesService,
+                new DummyFileUpdateMonitor(),
+                mock(UndoManager.class),
+                mock(StateManager.class),
+                mock(DialogService.class),
+                new CurrentThreadTaskExecutor()
+        ));
+        // Mock the behavior of getDuplicateDecision to return KEEP_RIGHT
+        Mockito.doReturn(decisionResult).when(importHandler).getDuplicateDecision(testEntry, duplicateEntry, bibDatabaseContext);
+
+        // Act
+        BibEntry result = importHandler.handleDuplicates(bibDatabaseContext, testEntry, duplicateEntry);
+
+        // Assert
+        assertFalse(bibDatabase.getEntries().contains(duplicateEntry)); // Assert that the duplicate entry was removed from the database
+        assertEquals(testEntry, result); // Assert that the original entry is returned
```

Comment:
For future PRs, please put single line comments above the statements. You can separate `// Assert` by an empty line, this is OK.